pkg/nas/controllerserver.go

```diff
@@ -341,36 +347,41 @@ func (cs *controllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVol
 	}
 
 	if volumeAs == "filesystem" {
-		log.Infof("DeleteVolume: Start delete mountTarget %s for volume %s", nfsServer, req.VolumeId)
-		if fileSystemId == "" {
-			return nil, fmt.Errorf("DeleteVolume: Volume: %s in filesystem mode, with filesystemId empty", req.VolumeId)
-		}
-		deleteMountTargetRequest := aliNas.CreateDeleteMountTargetRequest()
-		deleteMountTargetRequest.FileSystemId = fileSystemId
-		deleteMountTargetRequest.MountTargetDomain = nfsServer
-		deleteMountTargetResponse, err := cs.nasClient.DeleteMountTarget(deleteMountTargetRequest)
-		if err != nil {
-			log.Errorf("DeleteVolume: requestId[%s], volume[%s], fail to delete nas mountTarget %s: with %v", deleteMountTargetResponse.RequestId, req.VolumeId, nfsServer, err)
-			return nil, status.Error(codes.Internal, err.Error())
-		}
-		// remove the pvc mountTarget mapping if exist
-		if _, ok := pvcMountTargetMap[req.VolumeId]; ok {
-			delete(pvcMountTargetMap, req.VolumeId)
-		}
-		log.Infof("DeleteVolume: Volume %s MountTarget %s deleted successfully and Start delete filesystem %s", req.VolumeId, nfsServer, fileSystemId)
-
-		deleteFileSystemRequest := aliNas.CreateDeleteFileSystemRequest()
-		deleteFileSystemRequest.FileSystemId = fileSystemId
-		deleteFileSystemResponse, err := cs.nasClient.DeleteFileSystem(deleteFileSystemRequest)
-		if err != nil {
-			log.Errorf("DeleteVolume: requestId[%s], volume %s fail to delete nas filesystem %s: with %v", deleteFileSystemResponse.RequestId, req.VolumeId, fileSystemId, err)
-			return nil, status.Error(codes.Internal, err.Error())
-		}
-		// remove the pvc filesystem mapping if exist
-		if _, ok := pvcFileSystemIdMap[req.VolumeId]; ok {
-			delete(pvcFileSystemIdMap, req.VolumeId)
-		}
-		log.Infof("DeleteVolume: Volume %s Filesystem %s deleted successfully", req.VolumeId, fileSystemId)
+		if deletePolicy == "false" {
```

Comment:
deletePolicy is meaningless, maybe deleteEnsure is better?

or deletePolicy use selected value: "onlyPv"/"pvAndVolume".